### PR TITLE
parser,ast: parse ALTER TABLE .. TRUNCATE PARTITION

### PR DESCRIPTION
**Pull request by tiancaiamao**
Parse the ALTER TABLE .. TRUNCATE PARTITION syntax for TiDB

https://github.com/pingcap/tidb/pull/8624

### CODE REVIEW

File: parser.y [hunk 1]
```diff
@@ -2364,7 +2371,7 @@ DropViewStmt:
 |
 	"DROP" "VIEW" "IF" "EXISTS" TableNameList RestrictOrCascadeOpt
 	{
-		$$ = &ast.DropTableStmt{IfExists: true, Tables: $5.([]*ast.TableName), IsView: true}
+		$$ = &ast.DoStmt{}
```
Comment (kennytm) [inline parser.y]: Is this change expected?

Comment (crazycs520) [inline parser.y]: Why change this?

Comment (tiancaiamao) [reply parser.y]: Done

File: model/ddl.go [hunk 2]
```diff
@@ -80,6 +81,7 @@ var actionMap = map[ActionType]string{
 	ActionDropTablePartition:           "drop table partition",
 	ActionCreateView:                   "create view",
 	ActionModifyTableCharsetAndCollate: "modify table charset and collate",
+	ActionTruncateTablePartition:       "truncate table partition",
```
Comment (ciscoxll) [inline model/ddl.go]: s / `truncate table partition` / `alter table truncate partition` ?

**Pull request merged**